src/style/style.js

```diff
@@ -1365,6 +1365,10 @@ class Style extends Evented {
         this.light.updateTransitions(parameters);
     }
 
+    getTerrain() {
+        return this.terrain ? this.terrain.get() : this.stylesheet.terrain;
```

Comment:
I'm curious to why we return this.stylesheet.terrain when it's not set, the other root property getLight() does not adopt the same behavior. How do we detect whether terrain isn't set on a map?

Reply:
This implementation was based off my understanding of this line from the original issue `We should add map.getTerrain() which would return the serialized value previously set with setTerrain() or found in the stylesheet`.

>How do we detect whether terrain isn't set on a map?
My understanding is that `this.terrain` would only be defined is `setTerrain` has been called. Is that not correct?